packages/gotrue/lib/src/types/user.dart

```diff
@@ -54,54 +57,121 @@ class User {
     if (json['id'] == null) {
       return null;
     }
+
     return User(
-      id: json['id'] as String,
-      appMetadata: json['app_metadata'] as Map<String, dynamic>,
+      id: json['id'] ?? '',
+      appMetadata: json['app_metadata'] as Map<String, dynamic>? ?? {},
```

Comment:
```suggestion
      appMetadata: json['app_metadata'] ?? {},
```
Im pretty sure the `as Map` in the line below is unnecessary as well.

Reply:
Not really. `json['app_metadata']` may be a `Map<dynamic, dynamic>` when parsing from a parent json

Reply:
Let's just keep it! Either way it won't hurt!

Reply:
@bdlukaa That wouldn't work. You can't change the generics on `Map` via `as`, because these are separate types. [Here](https://zapp.run/edit/dart-z25w06ir25x0?entry=lib/main.dart&file=lib/main.dart) is a quick example.